File: Tutorials/Tuckshop/Server/Tuckshop.Domain/Tuckshop.Core.Models/Models/Orders/Queries/OrderLookup.cs
#pragma warning disable
namespace Tuckshop.Core.Models.Orders.Queries
{
  using System;
  using System.Collections;
  using System.Collections.Generic;
  using System.Linq;

  public class OrderLookup
  {
    public int OrderId { get; set; }
    public string CustomerName { get; set; }
    public DateTime OrderedOn { get; set; }
    public DateTime? CompletedOn { get; set; }
    public DateTime? CancelledOn { get; set; }
    public string CancelledReason { get; set; }
    public string CompletedBy { get; set; }
    public string CancelledBy { get; set; }
    public decimal OrderTotalExcl { get; set; }
    public decimal OrderTotal { get; set; }
    public object Items { get; set; }

    public OrderLookup WithDetails(IEnumerable<OrderDetailLookup> orderDetails)
    {
      this.OrderTotal = orderDetails.Sum(c => c.Value);
      this.OrderTotalExcl = orderDetails.Sum(c => c.Value - c.VAT);
      this.Items = orderDetails.ToList();
      return this;
    }
  }
}

File: Tutorials/Tuckshop/Client/Tuckshop.WebApp/src/Domain/Models/Orders/Queries/OrderLookup.ts
import { Attributes, List, LookupBase } from '@singularsystems/neo-core';
import OrderDetailLookup from './OrderDetailLookup';
// import OrderDetailLookup from './OrderDetailLookup';

export default class OrderLookup extends LookupBase {

    constructor() {
        super();
        this.makeBindable();
    }

    public readonly orderId: number = 0;

    public readonly customerName: string = "";

    @Attributes.Date()
    public readonly orderedOn: Date = new Date();
    
    @Attributes.Observable()
    @Attributes.Date()
    public readonly completedOn: Date | null = null;

    @Attributes.Observable()
    @Attributes.Date()
    public readonly cancelledOn: Date | null = null;

    public readonly cancelledReason: string = "";

    public readonly completedBy: string = "";

    public readonly cancelledBy: string = "";

    @Attributes.Float()
    public readonly orderTotalExcl: number = 0;

    @Attributes.Float()
    public readonly orderTotal: number = 0;

    public readonly items = new List(OrderDetailLookup);

    public readonly completedByFirstName: string | null = null;

    public readonly completedByLastName: string | null = null;

    public readonly cancelledByFirstName: string | null = null;

    public readonly cancelledByLastName: string | null = null;

    public get canAction(){
        return !this.completedOn && !this.cancelledOn;
    }
    // Client only properties / methods

    @Attributes.Observable()
    public isExpanded = false; // This property is used to control the expansion of the order details in the UI. It is not persisted to the server and is only relevant for the client-side view.
}